### PR TITLE
[11.0] [MIG] account_payment_mode

### PR DESCRIPTION
**Pull request by etobella**
Migration to 11.0

### CODE REVIEW

**Review by rgarnau — APPROVED** [review]
LGTM :+1:

**Review by JordiBForgeFlow — APPROVED** [review]
:+1: Technical and functional review

**Review by HviorForgeFlow — APPROVED** [review]
LGTM :+1:

File: account_payment_mode/models/account_journal.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    @api.constrains('company_id')
+    def company_id_account_payment_mode_constrains(self):
+        for journal in self:
+            mode = self.env['account.payment.mode'].search([
+                ('fixed_journal_id', '=', journal.id),
+                ('company_id', '!=', journal.company_id.id)], limit=1)
+            if mode:
+                raise ValidationError(_(
+                    "The company of the journal '%s', does not match "
```
Comment (elicoidal) [inline account_payment_mode/models/account_journal.py]: remove comma

File: account_payment_mode/models/account_journal.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                    "with the company of the payment mode '%s' where it is "
+                    "being used as Fixed Bank Journal.") % (
+                    journal.name, mode.name))
+            mode = self.env['account.payment.mode'].search([
+                ('variable_journal_ids', 'in', [journal.id]),
+                ('company_id', '!=', journal.company_id.id)], limit=1)
+            if mode:
+                raise ValidationError(_(
+                    "The company of the journal '%s', does not match "
```
Comment (elicoidal) [inline account_payment_mode/models/account_journal.py]: here too

File: account_payment_mode/models/account_payment_method.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+from odoo import models, fields, api
+
+
+class AccountPaymentMethod(models.Model):
+    _inherit = 'account.payment.method'
+
+    code = fields.Char(
+        string='Code (Do Not Modify)',
+        help="This code is used in the code of the Odoo module that handle "
```
Comment (elicoidal) [inline account_payment_mode/models/account_payment_method.py]: handles

File: account_payment_mode/models/account_payment_mode.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    bank_account_link = fields.Selection([
+        ('fixed', 'Fixed'),
+        ('variable', 'Variable'),
+        ], string='Link to Bank Account', required=True,
+        help="For payment modes that are always attached to the same bank "
+        "account of your company (such as wire transfer from customers or "
+        "SEPA direct debit from suppliers), select "
+        "'Fixed'. For payment modes that are not always attached to the same "
+        "bank account (such as SEPA Direct debit for customers, wire transfer "
+        "to suppliers), you should choose 'Variable', which means that you "
```
Comment (elicoidal) [inline account_payment_mode/models/account_payment_mode.py]: For consistency: choose ->select

File: account_payment_mode/models/account_payment_mode.py [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+                    mode.name, mode.fixed_journal_id.name))
+
+    @api.constrains('company_id', 'variable_journal_ids')
+    def company_id_variable_journal_ids_constrains(self):
+        for mode in self:
+            if any(mode.company_id != j.company_id for j in
+                   mode.variable_journal_ids):
+                raise ValidationError(_(
+                    "The company of the payment mode '%s', does not match "
+                    "with one of the Allowed Bank Journals.") % mode.name)
```
Comment (elicoidal) [inline account_payment_mode/models/account_payment_mode.py]: the one

**Review by elicoidal — APPROVED** [review]
Some nitpicking (non blocking)

**Comment by etobella** [comment]
@elicoidal changes applied.

**Comment by pedrobaeza** [comment]
Jordi, please don't merge without squashing useless commits like 5228214882dac8857f29cc147e698620604a1b75

**Pull request merged**